crates/torii/core/src/sql.rs

```diff
@@ -160,33 +183,37 @@ impl Sql {
         Ok(())
     }
 
-    pub fn set_metadata(&mut self, resource: &FieldElement, uri: &str) {
+    pub fn set_metadata(&mut self, resource: &FieldElement, uri: &str, block_timestamp: u64) {
         let resource = Argument::FieldElement(*resource);
         let uri = Argument::String(uri.to_string());
+        let executed_at = Argument::String(utc_dt_string_from_timestamp(block_timestamp));
 
         self.query_queue.enqueue(
-            "INSERT INTO metadata (id, uri) VALUES (?, ?) ON CONFLICT(id) DO UPDATE SET \
-             id=excluded.id, updated_at=CURRENT_TIMESTAMP",
-            vec![resource, uri],
+            "INSERT INTO metadata (id, uri, executed_at) VALUES (?, ?, ?) ON CONFLICT(id) DO \
+             UPDATE SET id=excluded.id, executed_at=excluded.executed_at, \
+             updated_at=CURRENT_TIMESTAMP",
+            vec![resource, uri, executed_at],
         );
     }
 
     pub async fn update_metadata(
         &mut self,
         resource: &FieldElement,
         uri: &str,
+        block_timestamp: u64,
         metadata: &WorldMetadata,
         icon_img: &Option<String>,
         cover_img: &Option<String>,
     ) -> Result<()> {
         let json = serde_json::to_string(metadata).unwrap(); // safe unwrap
 
-        let mut columns = vec!["id", "uri", "json"];
+        let mut columns = vec!["id", "uri", "executed_at", "json"];
```

Comment:
I don't think we need to set this here. I think it is already set in the `set_metadata` call in the processor. We should be able to revert this function body to the original implementation